src/ui/articleview.cc

```diff
@@ -674,63 +675,24 @@ bool ArticleView::eventFilter( QObject * obj, QEvent * ev )
 
 #else
   if ( ev->type() == QEvent::Gesture ) {
-    Gestures::GestureResult result;
-    QPoint pt;
-
-    bool handled = Gestures::handleGestureEvent( obj, ev, result, pt );
-
-    if ( handled ) {
-      if ( result == Gestures::ZOOM_IN )
-        emit zoomIn();
-      else if ( result == Gestures::ZOOM_OUT )
-        emit zoomOut();
-      else if ( result == Gestures::SWIPE_LEFT )
-        back();
-      else if ( result == Gestures::SWIPE_RIGHT )
-        forward();
-      else if ( result == Gestures::SWIPE_UP || result == Gestures::SWIPE_DOWN ) {
-        int delta        = result == Gestures::SWIPE_UP ? -120 : 120;
-        QWidget * widget = static_cast< QWidget * >( obj );
-
-        QPoint angleDelta( 0, delta );
-        QPoint pixelDetal;
-        QWidget * child = widget->childAt( widget->mapFromGlobal( pt ) );
-        if ( child ) {
-          QWheelEvent whev( child->mapFromGlobal( pt ),
-                            pt,
-                            pixelDetal,
-                            angleDelta,
-                            Qt::NoButton,
-                            Qt::NoModifier,
-                            Qt::NoScrollPhase,
-                            false );
-          qApp->sendEvent( child, &whev );
-        }
-        else {
-          QWheelEvent whev( widget->mapFromGlobal( pt ),
-                            pt,
-                            pixelDetal,
-                            angleDelta,
-                            Qt::NoButton,
-                            Qt::NoModifier,
-                            Qt::NoScrollPhase,
-                            false );
-          qApp->sendEvent( widget, &whev );
+    auto gestureEvent   = static_cast< QGestureEvent * >( ev );
+    auto activeGestures = gestureEvent->activeGestures();
+    for ( auto g : activeGestures ) {
+      if ( g->gestureType() == Qt::PanGesture && g->state() == Qt::GestureFinished ) {
+        auto panGesture = static_cast< QPanGesture * >( g );
+        auto offset     = panGesture->offset();
+        if ( qFabs( offset.x() ) > qFabs( offset.y() ) * 6 ) {
+          offset.x() > 0 ? back() : forward();
```

Comment:
The direction is swapped.

New version

Move figures from left edge toward right is going back.
Move figures from right edge toward left is going forward.

Original version:

Draw an arrow to left is going back
Draw an arrow to right is going forward

Both make sense, but the new version is consistent with mobile devices.